shell/shared/grain.js

```diff
@@ -1644,9 +1644,8 @@ Router.map(function () {
     path: "/apps/:appId",
     template: "appDetails",
     waitOn: function () {
-      return [
-        Meteor.subscribe("appIndex", this.params.appId),
-      ];
+      return globalSubs.concat(
```

Comment:
Does this not lead to duplicate subscriptions which are expensive to dedup server-side?